src/d/actor/d_a_obj_tntrap.cpp

```diff
@@ -4,172 +4,643 @@
-void daObjTnTrap_c::create_heap() {
-    /* Nonmatching */
+bool daObjTnTrap_c::create_heap() {
+    bool o_retval = true;
+    cBgD_t* pcVar1 = (cBgD_t*)dComIfG_getObjectRes(l_arcname,0x3);
+    field_0xD58 = dBgW_NewSet(pcVar1,cBgW::MOVE_BG_e,&field_0xD5C.calcMtx);
+    if(field_0xD58 == NULL){
+        o_retval = false;
+    }
+    return o_retval;
+
+    
 }
 
+
 /* 000003E4-000005F8       .text particle_set__13daObjTnTrap_cFif */
-void daObjTnTrap_c::particle_set(int, float) {
-    /* Nonmatching */
+void daObjTnTrap_c::particle_set(int i_particleId, float param_2) {
+
+   cXyz ball;
+
+    if(field_0xDE0[i_particleId] == 1){
+        if(field_0xDE4[i_particleId] != param_2){
+            particle_delete(i_particleId);
+        }else{
+            return;
+        }
+    }
+
+    for(int i = 0; i < 2; i++){
+        if(field_0xD5C.emitterPairs[i_particleId][i] == NULL){
+            ball.x = l_offset_ball[i][0];
+            ball.y = l_offset_ball[i][1]+param_2;
+            ball.z = l_offset_ball[i][2];
+            field_0xD5C.emitterPairs[i_particleId][i] = dComIfGp_particle_set(0x82EA,&home.pos,&shape_angle);
+            field_0xD5C.emitterPairs[i_particleId][i]->setEmitterTranslation(JGeometry::TVec3<f32>(ball.x,ball.y,ball.z));
+        }
+    }
+    for(int i = 0; i < 3; i++){
+        if(field_0xD5C.emitterPairs2[i_particleId][i] == NULL){
+            ball.x = l_offset_thunder[i][0];
+            ball.y = l_offset_thunder[i][1]+param_2;
+            ball.z = l_offset_thunder[i][2];
+            field_0xD5C.emitterPairs2[i_particleId][i] = dComIfGp_particle_set(0x82EB,&home.pos,&shape_angle);
+            field_0xD5C.emitterPairs2[i_particleId][i]->setEmitterTranslation(JGeometry::TVec3<f32>(ball.x,ball.y,ball.z));
+        }
+    }
+    field_0xDE4[i_particleId] = param_2;
+    field_0xDE0[i_particleId] = 1;
+    return;
 }
 
 /* 000005F8-000006A4       .text particle_delete__13daObjTnTrap_cFi */
-void daObjTnTrap_c::particle_delete(int) {
-    /* Nonmatching */
+void daObjTnTrap_c::particle_delete(int param_1) {
+
+    if(field_0xDE0[param_1] == 1){
+        int i;
+        for(i = 0; i < 2; i++){
+            if(field_0xD5C.emitterPairs[param_1][i] != 0){
+                field_0xD5C.emitterPairs[param_1][i]->becomeInvalidEmitter();
+                field_0xD5C.emitterPairs[param_1][i] = NULL;
+            }
+        }
+        for(i = 0; i < 3; i++){
+            if(field_0xD5C.emitterPairs2[param_1][i] != 0){
+                field_0xD5C.emitterPairs2[param_1][i]->becomeInvalidEmitter();
+                field_0xD5C.emitterPairs2[param_1][i] = NULL;
+            }
+        }
+    }
+    field_0xDE0[param_1] = 0;
+    return;
+    
 }
 
 /* 000006A4-0000072C       .text set_se__13daObjTnTrap_cFv */
 void daObjTnTrap_c::set_se() {
-    /* Nonmatching */
+
+    if(field_0xDC0 >= 5 || field_0xDC0 < 1){
+        return;
+    }
+    fopAcM_seStartCurrent(this,0x6239,0);
+
 }
 
+u32 fillerbytes[12] = {
```

Comment:
Is to make the data section match up? Check out this part of the tutorial, one of the weak data/bss includes might be able to fix it instead: https://github.com/zeldaret/tww/blob/main/docs/decompiling.md#missing-weak-data